kolibri/plugins/learn/assets/src/views/explore-page/index.vue

```diff
@@ -42,23 +49,66 @@
     $trs: {
       explore: 'Topics',
       navigate: 'Navigate content using headings',
+      all: 'All content',
+      topics: 'Topics',
+      exercises: 'Exercises',
+      videos: 'Videos',
+      audio: 'Audio',
+      documents: 'Documents',
+      html5: 'HTML5 Apps',
+      display: 'Display',
     },
     components: {
       'page-header': require('../page-header'),
       'topic-list-item': require('../topic-list-item'),
       'content-grid-item': require('../content-grid-item'),
       'card-grid': require('../card-grid'),
       'card-list': require('../card-list'),
+      'ui-select': require('keen-ui/src/UiSelect'),
     },
+    data: () => ({
+      selectedFilter: '',
+    }),
     computed: {
+      filterOptions() {
+        const options = [{ label: this.$tr('all'), value: 'all' }];
+        if (this.contentsContain(ContentNodeKinds.TOPIC)) {
+          options.push({ label: this.$tr('topics'), value: ContentNodeKinds.TOPIC });
+        }
+        if (this.contentsContain(ContentNodeKinds.EXERCISE)) {
+          options.push({ label: this.$tr('exercises'), value: ContentNodeKinds.EXERCISE });
+        }
+        if (this.contentsContain(ContentNodeKinds.VIDEO)) {
+          options.push({ label: this.$tr('videos'), value: ContentNodeKinds.VIDEO });
+        }
+        if (this.contentsContain(ContentNodeKinds.AUDIO)) {
+          options.push({ label: this.$tr('audio'), value: ContentNodeKinds.AUDIO });
+        }
+        if (this.contentsContain(ContentNodeKinds.DOCUMENT)) {
+          options.push({ label: this.$tr('documents'), value: ContentNodeKinds.DOCUMENT });
+        }
+        if (this.contentsContain(ContentNodeKinds.HTML5)) {
+          options.push({ label: this.$tr('html5'), value: ContentNodeKinds.HTML5 });
+        }
+        return options;
+      },
       title() {
         return this.isRoot ? this.$tr('explore') : this.topic.title;
       },
       subtopics() {
         return this.contents.filter(content => content.kind === ContentNodeKinds.TOPIC);
       },
+      filteredContents() {
+        if (this.selectedFilter.value === 'all') {
+          return this.contents;
+        }
+        return this.contents.filter(content => content.kind === this.selectedFilter.value);
+      },
     },
     methods: {
+      contentsContain(kind) {
```

Comment:
It might be more performant to use something like lodash `some`, since it will return true when it finds teh first element of `contents` satisfying the predicate (instead of traversing the entire array).